paddle/fluid/operators/recurrent_op.cc

```diff
@@ -208,6 +207,13 @@ void RecurrentOp::RunImpl(const framework::Scope &scope,
       *program, block->ID(), Attr<std::vector<std::string>>(
                                  kSkipEagerDeletionVars) /*skip_ref_cnt_vars*/);
 
+  static std::mutex mutex;
+  std::lock_guard<std::mutex> lock(mutex);
+  StepScopes scopes = CreateStepScopes(dev_ctx, scope, seq_len);
+  // TODO(gfwm2013) Function CreateStepScopes would make segmentation fault in
```

Comment:
You just wrote the problem but not the cause. We have found the reason. So please write down the reason of the segmentation fault. Then if you are going to do other things and forget it or some other software engineer takes place of you to fix here, we can still know the reason and debug easily.